doc/rados/troubleshooting/troubleshooting-osd.rst

```diff
@@ -267,206 +267,270 @@ If the cluster has started but an OSD isn't starting, check the following:
+this is the point at which it raises the ``OSD_NEARFULL`` health check.
+
+Different OSDs within a cluster will vary in how much data is allocated to them
+by Ceph. To check "fullness" by displaying data utilization for every OSD, run
+the following command:
+
+.. prompt:: bash
 
-Note that individual OSDs within a cluster will vary in how much data Ceph
-allocates to them.  This utilization can be displayed for each OSD with ::
+   ceph osd df
 
-	ceph osd df
+To check "fullness" by displaying a cluster’s overall data usage and data
+distribution among pools, run the following command:
 
-Overall cluster / pool fullness can be checked with ::
+.. prompt:: bash
 
-	ceph df 
+   ceph df 
 
-Pay close attention to the **most full** OSDs, not the percentage of raw space
-used as reported by ``ceph df``.  It only takes one outlier OSD filling up to
-fail writes to its pool.  The space available to each pool as reported by
-``ceph df`` considers the ratio settings relative to the *most full* OSD that
-is part of a given pool.  The distribution can be flattened by progressively
-moving data from overfull or to underfull OSDs using the ``reweight-by-utilization``
-command.  With Ceph releases beginning with later revisions of Luminous one can also
-exploit the ``ceph-mgr`` ``balancer`` module to perform this task automatically
-and rather effectively.
+When examining the output of the ``ceph df`` command, pay special attention to
+the **most full** OSDs, as opposed to the percentage of raw space used. If a
+single outlier OSD becomes full, all writes to this OSD's pool might fail as a
+result. When ``ceph df`` reports the space available to a pool, it considers
+the ratio settings relative to the *most full* OSD that is part of the pool. To
+flatten the distribution, two approaches are available: (1) Using the
+``reweight-by-utilization`` command to progressively move data from excessively
+full OSDs or move data to insufficiently full OSDs, and (2) in later revisions
+of Luminous and subsequent releases, exploiting the ``ceph-mgr`` ``balancer``
+module to perform the same task automatically.
 
-The ratios can be adjusted:
+To adjust the "fullness" ratios, run a command or commands of the following
+form:
+
+.. prompt:: bash
+
+   ceph osd set-nearfull-ratio <float[0.0-1.0]>
+   ceph osd set-full-ratio <float[0.0-1.0]>
+   ceph osd set-backfillfull-ratio <float[0.0-1.0]>
+
+Sometimes full cluster issues arise because an OSD has failed. This can happen
+either because of a test or because the cluster is small, very full, or
+unbalanced. When an OSD or node holds an excessive percentage of the cluster's
+data, component failures or natural growth can result in the ``nearfull`` and
+``full`` ratios being exceeded.  Anyone testing how Ceph reacts to OSD failures
+on a small cluster is advised to leave ample free disk space and to consider
+temporarily lowering the OSD ``full ratio``, OSD ``backfillfull ratio``, and
+OSD ``nearfull ratio``.
```

Comment:
When testing Ceph resilience to OSD failures on a small cluster, it is

Reply:
Accepted.